src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -850,13 +850,11 @@ InstanceKlass* SystemDictionaryShared::prepare_shared_lambda_proxy_class(Instanc
   assert(nest_host == shared_nest_host, "mismatched nest host");
 
   EventClassLoad class_load_start_event;
-  {
-    MutexLocker mu_r(THREAD, Compile_lock);
 
-    // Add to class hierarchy, and do possible deoptimizations.
-    SystemDictionary::add_to_hierarchy(loaded_lambda);
-    // But, do not add to dictionary.
-  }
+  // Add to class hierarchy, and do possible deoptimizations.
+  SystemDictionary::add_to_hierarchy(loaded_lambda, THREAD);
+  // But, do not add to dictionary.
```

Comment:
nit comment location: move above L855 like other places?

Reply:
I had already moved the other ones under instead.

Reply:
OK. Being consistent is the goal.